### PR TITLE
Convert some classes to record in Pinot

### PR DESCRIPTION
**Pull request by ebyhr**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Review by findepi — APPROVED** [review]
any non mechanical changes?

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryInfo.java [hunk 1]
```diff
@@ -13,76 +13,13 @@
...
-    public String toString()
+    public PinotQueryInfo
     {
-        return toStringHelper(this)
-                .add("query", query)
-                .add("table", table)
-                .add("groupByClauses", groupByClauses)
-                .toString();
+        requireNonNull(table, "table is null");
+        requireNonNull(query, "query is null");
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryInfo.java:23]: Added requireNonNull. These fields are non-null as far as I confirmed.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/OrderByExpression.java [hunk 2]
```diff
@@ -13,66 +13,12 @@
...
-
-    @Override
-    public String toString()
+    public OrderByExpression
     {
-        return toStringHelper(this)
-                .add("expression", expression)
-                .add("asc", asc)
-                .toString();
+        requireNonNull(expression, "expression is null");
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/OrderByExpression.java:22]: Fixed the error message. It was `column is null` before this change.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTable.java [hunk 3]
```diff
@@ -13,203 +13,55 @@
...
+        orderBy = ImmutableList.copyOf(requireNonNull(orderBy, "orderBy is null"));
+        requireNonNull(limit, "limit is null");
+        requireNonNull(offset, "offset is null");
+        queryOptions = ImmutableMap.copyOf(requireNonNull(queryOptions, "queryOptions is null"));
+        requireNonNull(query, "query is null");
     }
 
-    @JsonProperty
-    public Optional<String> getHavingExpression()
+    public boolean aggregateInProjections()
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTable.java:62]: Renamed the method from `isAggregateInProjections` and moved the logic from a constructor.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/AggregateExpression.java [hunk 4]
```diff
@@ -42,70 +39,13 @@ private static String stripDoubleQuotes(String expression)
...
-    }
-
-    @JsonProperty
-    public boolean isReturnNullOnEmptyGroup()
-    {
-        return returnNullOnEmptyGroup;
-    }
-
-    public String toFieldName()
+    public String fieldName()
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/AggregateExpression.java:42]: Renamed the method from `toFieldName`.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/AggregateExpression.java [hunk 5]
```diff
@@ -42,70 +39,13 @@ private static String stripDoubleQuotes(String expression)
...
     {
         return format("%s(%s)", function, argument);
     }
 
-    public String toExpression()
+    public String expression()
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/AggregateExpression.java:47]: Renamed the method from `toExpression`.

**Review by findepi — APPROVED** [review]

**Pull request merged**